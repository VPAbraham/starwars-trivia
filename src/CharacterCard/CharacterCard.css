.div--character-card {
  border: 2px solid;
  font-size: 1.2rem;
  height: 65vh;
  overflow-y: scroll;
  text-align: justify;
  width: 30vw;
}

.h3--character-card {
  display: flex;
  justify-content: space-between;
  margin: 15px;
}

.p--char-data {
  text-decoration: underline;
}

.li--char-films {
  padding: 2px;
}