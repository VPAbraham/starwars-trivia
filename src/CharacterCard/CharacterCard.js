import React from 'react';
import './CharacterCard.css';
import favorite from '../images/lightsaber.png';
import notFav from '../images/lightsaber-disabled.png';
import propTypes from 'prop-types';


const CharacterCard = (props) => {
  const { name, homeworld, species, relatedFilms, isFavorite } = props
  const films = relatedFilms.map(film => <li className='li li--char-films'>{film}</li>)
  const favIcon = !isFavorite ? notFav : favorite;
  return (
    <div className='div div--character-card'>
      <h3 className='h3 h3--character-card'><img className='img--fav-icon' src={favIcon} onClick={event => props.method({ name, homeworld, species, relatedFilms, isFavorite })}/>{name}</h3>
      <ul>
        <p className='p p--char-data'>Homeworld</p>
        <li>{homeworld[0]}</li>
        <li>Population: {homeworld[1]}</li>
        <p className='p p--char-data'>Species</p>
        <li>{species}</li>
        <p className='p p--char-data'>Appears in:</p>
        <ul>{films}</ul>
      </ul>  
    </div>
  )
}

export default CharacterCard;

CharacterCard.propTypes = {
  name: propTypes.string,
  homeworld: propTypes.string,
  homeworldPop: propTypes.number,
  species: propTypes.string,
  relatedFilms: propTypes.array,
  isFavorite: propTypes.bool
}