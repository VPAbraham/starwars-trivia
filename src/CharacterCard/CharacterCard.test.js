import React from 'react';
import { shallow } from 'enzyme';
import CharacterCard from './CharacterCard'


describe('CharacterCard', () => {
  let wrapper
  const addFav = jest.fn()
  beforeEach(() => {
    wrapper = shallow(<CharacterCard
      name='Chris'
      homeworld={['Earth', 7000000000]}
      species='Human'
      relatedFilms={['Yup', 'Yuppers']}
      isFavorite={true}
      method={addFav}
    />
    )
  })
  it('should match the snapshot', () => {
    expect(wrapper).toMatchSnapshot()
  });

  it('should run addFav on click', () => {
    wrapper.find('img').simulate('click')
    expect(addFav).toHaveBeenCalled();
  });
})