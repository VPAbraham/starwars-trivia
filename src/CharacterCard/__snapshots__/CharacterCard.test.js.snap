// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`CharacterCard should match the snapshot 1`] = `
<div
  className="div div--character-card"
>
  <h3
    className="h3 h3--character-card"
  >
    <img
      className="img--fav-icon"
      onClick={[Function]}
      src="lightsaber.png"
    />
    Chris
  </h3>
  <ul>
    <p
      className="p p--char-data"
    >
      Homeworld
    </p>
    <li>
      Earth
    </li>
    <li>
      Population: 
      7000000000
    </li>
    <p
      className="p p--char-data"
    >
      Species
    </p>
    <li>
      Human
    </li>
    <p
      className="p p--char-data"
    >
      Appears in:
    </p>
    <ul>
      <li
        className="li li--char-films"
        key="Chris"
      >
        Yup
      </li>
      <li
        className="li li--char-films"
        key="Chris"
      >
        Yuppers
      </li>
    </ul>
  </ul>
</div>
`;
